api/python/provisioner/commands/destroy_vm.py

```diff
@@ -0,0 +1,157 @@
+                    f"ssh {node} yum erase -y"
+                    "cortx-prvsnr python36-cortx-prvsnr")
+
+    def _remove_passwordless_ssh(self, nodes):
+        for node in nodes:
+            logger.info(f"Remove passwordless configuration with {node}")
+            run_subprocess_cmd(
+                f"rm -rf /root/.ssh")
+
+    def _remove_salt(self, nodes):
+        salt_cache_dir = '/var/cache/salt'
+        salt_config = '/etc/salt'
+
+        for node in nodes:
+            logger.info(f"stopping salt-minion/salt-master on {node}")
+            run_subprocess_cmd(
+                f"ssh {node} systemctl stop salt-minion salt-master")
+
+            logger.info(f"remove salt rpms on {node}")
+            run_subprocess_cmd(
+                f"ssh {node} yum erase -y salt-minion salt-master salt")
+
+            logger.info(f"Removing salt from node {node}")
+            run_subprocess_cmd(f"ssh {node} rm -rf {salt_cache_dir}")
+            run_subprocess_cmd(f"ssh {node} rm -rf {salt_config}")
+
+    def _remove_dir(self, list_dir, nodes):
+        for node in nodes:
+            for di in list_dir:
+                logger.info(f"cleaning up {di} dir on {node}")
+                run_subprocess_cmd(f"ssh {node} rm -rf {di}")
+
+    def _run_states(self, states_group: str, run_args: run_args_type):
+        setup_type = run_args.setup_type
+        targets = run_args.targets
+        states = deploy_states[states_group]
+        stages = run_args.stages
+        primary = self._primary_id()
+        secondaries = f"not {primary}"
+        nodes = []
+
+        if states_group == 'prvsnr' and targets == ALL_MINIONS:
+            node_list = PillarKey('cluster/node_list')
+            pillar = PillarResolver(config.LOCAL_MINION).get([node_list])
+            pillar = next(iter(pillar.values()))
+            nodes = pillar[node_list]
+        else:
+            nodes.append(targets)
+
+        # apply states
+        for state in states:
+            if setup_type == SetupType.SINGLE:
+                self._apply_state(f"components.{state}", primary, stages)
+            else:
+                self._apply_state(f"components.{state}", targets, stages)
+
+        if states_group == 'prvsnr':
+            list_dir = []
+            list_dir.append(str(config.profile_base_dir().parent))
+            list_dir.append(config.CORTX_ROOT_DIR)
+            self._remove_prvsnr(nodes)
+            self._remove_dir(list_dir, nodes)
+            self._remove_salt(nodes)
+            self._remove_passwordless_ssh(nodes)
+
+    def run(self, **kwargs):  # noqa: C901
+        run_args = self._run_args_type(**kwargs)
+
+        if self._is_hw():
+            raise errors.ProvisionerError(
+                "The command is specifically for VM teardown. "
+            )
+
+        if run_args.states is None:  # all states
+            self._run_states('prvsnr', run_args)
+
+        else:
+            if 'prvsnr' in run_args.states:
```

Comment:
why we consider other values here? if it is going to be expanded in future please add `else` block where `NotImplementedError` is raised

Reply:
tomorrow I will start writing logic for remaining components, till now this is partial logic only for teardown of provisioner setup.

Reply:
For other components, we can wait.
Let's cleanup existing teardown first with existing shell script may be.

Reply:
Added NotImplementedError